bundles/org.openhab.binding.sncf/src/main/java/org/openhab/binding/sncf/internal/discovery/SncfDiscoveryService.java

```diff
@@ -0,0 +1,110 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.sncf.internal.discovery;
+
+import static org.openhab.binding.sncf.internal.SncfBindingConstants.*;
+
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.sncf.internal.SncfException;
+import org.openhab.binding.sncf.internal.dto.PlaceNearby;
+import org.openhab.binding.sncf.internal.handler.SncfBridgeHandler;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.i18n.LocationProvider;
+import org.openhab.core.library.types.PointType;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SncfDiscoveryService} searches for available
+ * station discoverable through API
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@Component(service = ThingHandlerService.class)
+@NonNullByDefault
+public class SncfDiscoveryService extends AbstractDiscoveryService implements ThingHandlerService {
+    private static final int SEARCH_TIME = 10;
+
+    private final Logger logger = LoggerFactory.getLogger(SncfDiscoveryService.class);
+
+    private @Nullable LocationProvider locationProvider;
+    private @Nullable SncfBridgeHandler bridgeHandler;
+
+    private int searchRange = 1500;
+
+    @Activate
+    public SncfDiscoveryService() {
+        super(SUPPORTED_THING_TYPES_UIDS, SEARCH_TIME);
+    }
+
+    @Override
+    public void deactivate() {
+        super.deactivate();
+    }
+
+    @Override
+    public void startScan() {
+        SncfBridgeHandler handler = bridgeHandler;
+        LocationProvider provider = locationProvider;
+        if (provider != null && handler != null) {
+            PointType location = provider.getLocation();
+            if (location != null) {
+                ThingUID bridgeUID = handler.getThing().getUID();
+                searchRange += 500;
+                try {
+                    List<PlaceNearby> places = handler.discoverNearby(location, searchRange);
+                    if (places != null && !places.isEmpty()) {
+                        places.forEach(place -> {
+                            String placeId = place.id;
+                            String thingId = placeId.replace(":", "_").replace("-", "_").replace("stop_point_", "");
+                            thingDiscovered(
+                                    DiscoveryResultBuilder.create(new ThingUID(STATION_THING_TYPE, bridgeUID, thingId))
+                                            .withLabel(place.stopPoint.name).withBridge(bridgeUID)
```

Comment:
You can have several places with the same `place.stopPoint.name`, like a bus station, a train station, ...
Maybe your label could include the kind of station ?

Alternative: having different thing types for each kind of stations ?

Reply:
The trick is apparently the kind of stop_point is the last element of the placeId (I did not found any alternate API call that could provide this information in a clean way) - but there is no 100% certainty that it is always true. So I was not happy to pull this information in the label.

Reply:
Your new code is better.